### ISSUE TITLE
[a11y]: Radio button is not selectable using keyboard when there is a single option

### ISSUE THREAD

**Issue by JeenaJose123** [carbon-design-system/carbon#19031]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox

### Operating System

MacOS

### Package version

1.72.0

### React version

18.3.1

### Automated testing tool and ruleset

Manual testing

### Description

Unable to selection the radio button using the keyboard when there is a single option.

### Reproduction/example

https://stackblitz.com/edit/xajngfwm?file=src%2FApp.jsx

### Steps to reproduce

create a radio button group with single radio button
try to select the radio button using tab key

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Gururajj77** [comment]
Hey @JeenaJose123 , I have checked the stackblitz, it's working as expected, could you tell me how you are interacting with the radio-button using keyboard?

**Comment by JeenaJose123** [comment]
I have used `Tab` key to move to radio button then click on `Enter` key to select the option. If I have multiple options I am able to select the options using `left/right` arrow keys. If there a single option nothing works for me.

**Comment by Gururajj77** [comment]
Hey @JeenaJose123 , I think you shouldn't use a single radio button for selection. Here is the [usage](https://carbondesignsystem.com/components/radio-button/usage/) doc for radio buttons. Meanwhile you can use single checkbox or toggle button for this purpose.

**Comment by JeenaJose123** [comment]
It's not like that there is always one option. There can be multiple options and this one is only an edge case with single option.

**Comment by mariyageorge01** [comment]
> I have used Tab key to move to radio button then clicked on Enter key to select the option. 

You can use `tab` key to move into radio button group and the press `space` to select an option. See https://carbondesignsystem.com/components/radio-button/usage/#interactions

**Comment by JeenaJose123** [comment]
yes it's selectable with `space`. Thank you @mariyageorge01
